uswds/uswds-site
# USWDS-Site - Dependencies: Use ruby 3.2.5

# Summary

Update local ruby version to match in CircleCI config [`.circleci/config.yml`]. Also replaces deprecated class for used in `jekyll_get.rb`.

## Related issue

n/a - This is a hotfix.

## Preview link

Any page that pulls from USWDS markdown files.

Preview links:
- [Documentation](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/jm-update-ruby-3.2.5/documentation/developers/)
- [Security](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/jm-update-ruby-3.2.5/about/security/)

## Problem statement

HTMLProofer errors were being reported on MD files pulled from USWDS repo. This is becaus our `jekyll_get` plugin was using a deprecated class to check if file existed.


## Solution

Updated local ruby version to match the one in CircleCI config.

https://github.com/uswds/uswds-site/blob/ab171bcfef790f5281209a0d318566105cc99b2f/.ruby-version#L1

https://github.com/uswds/uswds-site/blob/e7cb8b9940c0535d0818e5fff4cf1533f8217843/.circleci/config.yml#L10

```diff
- if not File.exists?(path)
+ if not File.exist?(path) 
```

## Testing and review

1. Pull `jm-update-ruby-3.2.5`
2. Update ruby, example using RVM `rvm install 3.2.5 --with-openssl-dir=$(brew --prefix openssl)`
3. Run `npm install & bundle install`
4. Run `npm start`

You **shouldn't** be seeing any errors and all checks should pass. Pages will render correctly too.
